src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestGenieTelemetryService.kt

```diff
@@ -92,5 +101,58 @@ class TestGenieTelemetryService(_project: Project) {
         File(telemetryFileName).bufferedWriter().use { out -> out.write(json) }
     }
 
-    class ModifiedTestCase(val original: String, val modified: String)
+    abstract class AbstractModifiedTestCase(val original: String, val modified: String)
+
+    class ModifiedTestCase(original: String, modified: String) : AbstractModifiedTestCase(original, modified) {
+
+        /**
+         * Calculate the differences in the assertions of the original and modified test code,
+         * and convert this ModifiedTestCase to a ModifiedTestCaseWithAssertions.
+         *
+         * @param project the current project
+         * @return a ModifiedTestCaseWithAssertions
+         */
+        internal fun convertToModifiedTestCaseWithAssertions(project: Project): ModifiedTestCaseWithAssertions {
+            val originalTestAssertions = extractAssertions(original, project)
+            val modifiedTestAssertions = extractAssertions(modified, project)
+            val removedAssertions = originalTestAssertions.minus(modifiedTestAssertions)
+            val addedAssertions = modifiedTestAssertions.minus(originalTestAssertions)
+
+            return ModifiedTestCaseWithAssertions(
+                this.original,
+                this.modified,
+                removedAssertions,
+                addedAssertions
+            )
+        }
+
+        /**
+         * Extracts assertions from a method.
+         *
+         * @param testCode the source code of the test
+         * @param project the currently open project
+         * @return the set of found assertion
+         */
+        private fun extractAssertions(testCode: String, project: Project): Set<String> {
+            val testClass: PsiClass = PsiElementFactory.getInstance(project).createClass("Test")
+            val psiMethod: PsiMethod =
+                PsiElementFactory.getInstance(project).createMethodFromText(testCode.trim(), testClass)
+
+            val allMethodCalls = psiMethod.body?.children
+                ?.filterIsInstance<PsiExpressionStatement>()
+                ?.map { it.firstChild }
+                ?.filterIsInstance<PsiMethodCallExpression>() ?: listOf()
+            val assertions = allMethodCalls.filter { it.firstChild.text.contains("assert") }
```

Comment:
Why do you use `firstChild` here?

Reply:
I was looking at the structure of the `PsiExpression` class in the debugger, and basically the first child is some sort of a reference expression that has what we need, and the last child contains something else